src/main/kotlin/org/rust/ide/refactoring/changeSignature/impl.kt

```diff
@@ -98,7 +98,9 @@ fun changeVisibility(function: RsFunction, config: RsChangeFunctionSignatureConf
 
     val vis = config.visibility
     if (vis != null) {
-        function.addBefore(vis, function.firstChild)
+        val firstChild = function.firstChild
+        val anchor = firstChild.takeIf { it !is RsAttr } ?: firstChild.getNextNonWhitespaceSibling()
```

Comment:
It doesn't work for this case:

```rust
// qwerty
fn foo() {
    todo!()
}
```

Note: `ChangeVisibilityIntention.makePublic` can be useful.

Reply:
I looked at the implementation, but thought that handling attributes is enough. Well, it isn't :) Thanks. I reused the metod from `ChangeVisibilityIntention`. Not sure if it should be placed in some other place.